### PR TITLE
Codec2 0.5 (new formula)

### PR DESCRIPTION
**Pull request by ast**
Codec 2 is an open source speech codec designed for communications
quality speech between 700 and 3200 bit/s. The main application is low
bandwidth HF/VHF digital radio. It fills a gap in open source,
free-as-in-speech voice codecs beneath 5000 bit/s and is released under
the GNU Lesser General Public License (LGPL).

### CODE REVIEW

File: Library/Formula/codec2.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Codec2 < Formula
+  desc "Codec 2 is an open source speech codec."
```
Comment (bfontaine) [inline Library/Formula/codec2.rb]: You can shorten this: `"Open source speech codec"`.

File: Library/Formula/codec2.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "http://www.rowetel.com/blog/?page_id=452"
+  url "http://files.freedv.org/codec2/codec2-0.5.tar.xz"
+  sha256 "1ffda04ec6629f5ad5a38349c6d9d38d29bfbc1c677c12014ff20d480a343f17"
+
+  depends_on "cmake" => :build
+
+  def install
+    Dir.mkdir "build_osx"
+    Dir.chdir "build_osx" do
```
Comment (bfontaine) [inline Library/Formula/codec2.rb]: Use `mkdir "build_osx" do` instead of `Dir.mkdir` and `Dir.chdir`.

**Comment by bfontaine** [comment]
The test is failing:
```
==> /usr/local/Cellar/codec2/0.5/bin/c2demo
[0] r = 16838
[1] r = 5758
[2] r = 10113
[3] r = 17515
[4] r = 31051
[5] r = 5627
[6] r = 23010
[7] r = 7419
[8] r = 16212
[9] r = 4086
usage: /usr/local/Cellar/codec2/0.5/bin/c2demo InputRawSpeechFile OutputRawSpeechFile
Error: codec2: failed
```

**Comment by ast** [comment]
The problems should be fixed now.

**Comment by DomT4** [comment]
```
== /usr/local/Library/Formula/codec2.rb ==
C: 18: 87: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

==> audit problems
codec2:
 * Use `system "#{bin}/c2enc", "2400", "test.raw", "test.c2"` instead of `system "#{bin}/c2enc 2400 test.raw test.c2"` 
```

File: Library/Formula/codec2.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Codec2 < Formula
+  desc "Open source speech codec."
+  homepage "http://www.rowetel.com/blog/?page_id=452"
+  url "http://files.freedv.org/codec2/codec2-0.5.tar.xz"
```
Comment (bfontaine) [inline Library/Formula/codec2.rb]: You can use `https://` for this `url`.

File: Library/Formula/codec2.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    mkdir "build_osx" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    # 8 bytes of raw audio data (silence).
+    (testpath/"test.raw").write([0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00].pack('C*'))
+    system "#{bin}/c2enc 2400 test.raw test.c2"
```
Comment (bfontaine) [inline Library/Formula/codec2.rb]: Use `system "#{bin}/c2enc", "2400", "test.raw", "test.c2"`

**Comment by ast** [comment]
Ok, should be fixed.

Sorry about all this, it's my first formula.

**Comment by bfontaine** [comment]
A few more style issues:

```
== Library/Formula/codec2.rb ==
C: 18: 87: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 19: 35: Unnecessary spacing detected.
C: 19: 48: Unnecessary spacing detected.
1 file inspected, 3 offenses detected
```

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @ast; we appreciate it! :tada:

**Pull request closed without merging**